components/patterns/elements/image-loader.tsx

```diff
@@ -0,0 +1,15 @@
+const ImageLoader = ({ src, width }) => {
+  const imageStyles = [
+    {w: 128, s: 'breakpoint_sm_2x'},
+    {w: 384, s: 'breakpoint_md_2x'},
+    {w: 400, s: 'breakpoint_md_2x'},
+    {w: 750, s: 'breakpoint_lg_2x'},
+    {w: 1200, s: 'breakpoint_xl_2x'},
+    {w: 3840, s: 'breakpoint_2xl_2x'},
+  ];
+  const style = imageStyles.find(s => s.w === width)?.s || 'breakpoint_xl_2x';
+  // Switch the image style for the desired width. Add the width parameter just to prevent console logs.
+  // https://nextjs.org/docs/messages/next-image-missing-loader-width
+  return src.toString().replace(/styles\/\w+\//, `styles/${style}/`) + `&w=${width}`;
```

Comment:
However unlikely, in theory, the src string could already have a `w` parameter. It would be safer to put the URL through  `new URL` and then set the searchParams instead of concatenating the string.

eg: https://nextjs.org/docs/app/api-reference/next-config-js/images#fastly

Reply:
I thought about that too. It would be more bullet proof that way, but since we're only ever pulling images from Drupal, there shouldn't ever be that param.